src/client/clientmap.cpp

```diff
@@ -218,17 +223,23 @@ void ClientMap::updateDrawList()
 				continue;
 			}
 
-			float range = 100000 * BS;
-			if (!m_control.range_all)
-				range = m_control.wanted_range * BS;
+			v3s16 block_coord = block->getPos();
+			v3s16 block_position = block_coord * MAP_BLOCKSIZE + MAP_BLOCKSIZE / 2;
```

Comment:
MapBlocks have their "node pos" precalculated so you'll want this to be:
```suggestion
			v3s16 block_position = block->getPosRelative() + MAP_BLOCKSIZE / 2;
```


But even better you could divide `cam_pos_nodes` by 16 for use by the first range check (L231), making this variable unnecessary.